### PR TITLE
get-secret-from-clusternamespace

### PR DESCRIPTION
**Pull request by calvix**
look for resources in the cluster namespace (instead of looking to default only)
 remove code for AWSConfig
removed unused code

### CODE REVIEW

File: pkg/giantnetes/utils.go [hunk 1]
```diff
@@ -98,27 +99,19 @@ func (u *Utils) GetTenantClusters(ctx context.Context, backup v1alpha1.ETCDBacku
...
-		}
 	case awsCAPI:
 		{
 			// Cluster API AWS backups are always supported.
 			return true, nil
 		}
 	case azure:
 		{
-			crd, err := crdClient.ProviderV1alpha1().AzureConfigs(crdNamespace).Get(ctx, cluster.clusterID, getOpts)
+			crd, err := crdClient.ProviderV1alpha1().AzureConfigs(cluster.clusterNamespace).Get(ctx, cluster.clusterID, getOpts)
```
Comment (whites11) [inline pkg/giantnetes/utils.go:114]: I'm afraid this won't work. there is some confusion in azure in terms of namespaces. Some CRs are in the org namespace, some are not. Considering the secret so far was looked for in the default namespace, it means it's there in azure.

Comment (calvix) [reply pkg/giantnetes/utils.go:114]: oh so Crs are in org namespace but the secret is still in default?

Comment (whites11) [reply pkg/giantnetes/utils.go:114]: Most CRs are in org, but the `AzureConfig` (legacy one) is still in default and all secrets are still in default
```
$ kubectl get cluster -n org-giantswarm
NAME    PHASE
5jka7   
c68pn   
$ kubectl get azureconfig
NAME    AGE
5jka7   545d
c68pn   666d
$ kubectl get secret -l giantswarm.io/cluster=5jka7
NAME                           TYPE     DATA   AGE
5jka7-api                      Opaque   3      545d
5jka7-app-operator-api         Opaque   3      545d
5jka7-calico-etcd-client       Opaque   3      545d
5jka7-certificate-encryption   Opaque   2      545d
5jka7-cluster-operator-api     Opaque   3      545d
5jka7-encryption               Opaque   1      545d
5jka7-etcd                     Opaque   3      545d
5jka7-node-operator            Opaque   3      545d
5jka7-prometheus               Opaque   3      545d
5jka7-service-account          Opaque   3      545d
5jka7-worker                   Opaque   3      545d
```

Comment (calvix) [reply pkg/giantnetes/utils.go:114]: but that mean it should work, if AzureConfig is in default and secrets are there as well

Comment (whites11) [reply pkg/giantnetes/utils.go:114]: it seems etcd backup operator is looking at `Cluster` not `AzureConfig`, isn't it?

Comment (calvix) [reply pkg/giantnetes/utils.go:114]:
```
// Azure
	{
		crdList, err := crdCLient.ProviderV1alpha1().AzureConfigs(metav1.NamespaceAll).List(ctx, listOpt)
		if err == nil {
			anySuccess = true
			for _, azureConfig := range crdList.Items {
				// Only backup cluster if it was not marked for delete.
				if azureConfig.DeletionTimestamp == nil {
					clusterList = append(clusterList, Cluster{azureConfig.Name, azureConfig.Namespace, azure})
				}
			}
		} else {
			u.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Error listing AzureConfigs: %s", err))
		}
	}
```

Comment (whites11) [reply pkg/giantnetes/utils.go:114]: ok sounds good then, thanks

**Review by whites11 — APPROVED** [review]

**Pull request merged**